text-tool/scripts/text-tool-2.0.0b.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Text tool
+ * 
+ * @version 2.0.0b
+ */
+
+var text;
+(function () {
+  function wrap(text, before, after) {
+    return (before || '') + text + (after || before || '');
+  }
+
+  function doOnChars(text,operationOnChars) {
+    let newText = '';
+    for (let index in text) {
+      newText += operationOnChars(text[index]);
+    }
+    return newText;
+  }
+
+  function charInBrackets(char) {
+    return char !== " " ? "[" + char + "]" : char;
+  }
+
+  text = function (_text) {
+    var __text = new String(_text);
+
+    __text.wrap = function (before, after) {
+      return text(wrap(_text, before, after));
+    }
+
+    __text.doOnChars = function () {
+      return text(doOnChars(_text,charInBrackets));
```

Comment:
WARN! doOnChars SHOULD accept a callback function, and NOT apply an specific one. `charInBrackets` it's just a good example for testing, but should not be here in the text-tool impl. `doOnChars` should by a higher other function.